content/2015.10.01.outreachy.rst

```diff
@@ -0,0 +1,120 @@
+
+.. epigraph::
+
+  By having a program targeted specifically towards women, we found that we
+  reached talented and passionate participants, who were uncertain about how to
+  start otherwise.
+
+  We are now expanding the program to people from more groups underrepresented
+  in FOSS. We hope this effort will help many people learn how exciting, varied
+  and valuable work on FOSS projects can be and how inclusive the community
+  really is.
+
+  -- `The Gnome outreachy program wiki <https://wiki.gnome.org/Outreachy>`_
+
+In our team, and at Mozilla in general, we thrive to build an inclusive
+community. And we believe being more inclusive starts by reaching to the people
+who are often under-represented in tech.
+
+That's why we are participating in the gnome Outreachy program this year by
+proposing an internship around Kinto and making instances discoverable.
+
+Check out the eligibility details at the end of the article if you're
+interested!
+
+Making Kinto instances discoverable
+===================================
+
+The `Kinto project <https://kinto.readthedocs.org>`_ aims to bring storage
+instances to everyone, attached to their
+`Firefox Accounts <https://accounts.firefox.com>`_ (Well, it actually supports
+multiple authentication schemes, but FxA is integrated with it, and that is
+part of the solution we want to deliver).
+
+Currently, Kinto is thought as a centralized server: there is one instance, and
+everyone authenticates on this instance. Items are shared between users of a
+same instance.
+
+This doesn't resonates well with multiple goals we have: scalability is harder
+when there is one endpoint, and it's also not interoperable.
+
+For instance, imagine Alice and Bob. Bob is using Mozilla's servers to store
+his data, whereas Alice deployed her own Kinto instance.
+
+There are different use cases:
+
+* Alice wants to use `Routina <https://github.com/leplatrem/routina>`_, an
+  application that stores its data inside a Kinto instance. As such, Routina
+  needs a way to discover where it should store its data, and send the requests
+  to this server;
+* Bob and Alice want to collaborate on a set of data (think about a shared
+  expenses webapp). There should be a way for Alice to host everything and
+  grant access to Bob to her data. The webapp should be able to use the
+  correct server.
+
+Here are the different steps that could allow these scenarios:
+
+* At the moment they authenticate, the client detects the email address used,
+  and relies on the domain part to do a
+  `Web Finger <https://tools.ietf.org/html/rfc7033>`_ request on the domain
+  (*) and for the specified user.
+* In case the identified server doesn't support WebFinger, it uses a central
+  repository to lookup where the Kinto server is located.
+* Once the Kinto server located, all requests should be issued against this
+  server.
+
+(*) It is also possible to use the same mechanism to discover the FxA endpoints.
```

Comment:
Here we should say that there are work in progress to do so using OpenId providers to log into a Firefox Account and that this will probably be using WebFinger as well.